pkg/trace/stats/otel_util.go

```diff
@@ -73,10 +73,25 @@ func OTLPTracesToConcentratorInputsWithObfuscation(
 		if _, exists := ignoreResNames[resourceName]; exists {
 			continue
 		}
-		env := traceutil.GetOTelEnv(otelres)
-		hostname := traceutil.GetOTelHostname(otelspan, otelres, conf.OTLPReceiver.AttributesTranslator, conf.Hostname)
-		version := traceutil.GetOTelAttrValInResAndSpanAttrs(otelspan, otelres, true, string(semconv.ServiceVersionKey))
-		cid := traceutil.GetOTelAttrValInResAndSpanAttrs(otelspan, otelres, true, string(semconv.ContainerIDKey), string(semconv.K8SPodUIDKey))
+
+		sattr := otelspan.Attributes()
+		rattr := otelres.Attributes()
+		env := traceutil.GetOTelAttrFromEitherMap(sattr, rattr, true, transform.KeyDatadogEnvironment)
+		if env == "" && !conf.OTLPReceiver.IgnoreMissingDatadogFields {
+			env = traceutil.GetOTelEnv(otelspan, otelres)
+		}
+		hostname := traceutil.GetOTelAttrFromEitherMap(sattr, rattr, true, transform.KeyDatadogHost)
+		if hostname == "" && !conf.OTLPReceiver.IgnoreMissingDatadogFields {
+			hostname = traceutil.GetOTelHostname(otelspan, otelres, conf.OTLPReceiver.AttributesTranslator, conf.Hostname)
+		}
+		version := traceutil.GetOTelAttrFromEitherMap(sattr, rattr, true, transform.KeyDatadogVersion)
+		if version == "" && !conf.OTLPReceiver.IgnoreMissingDatadogFields {
+			version = traceutil.GetOTelAttrFromEitherMap(sattr, rattr, true, string(semconv.ServiceVersionKey))
+		}
+		cid := traceutil.GetOTelAttrFromEitherMap(sattr, rattr, true, transform.KeyDatadogContainerID)
+		if cid == "" && !conf.OTLPReceiver.IgnoreMissingDatadogFields {
+			cid = traceutil.GetOTelAttrFromEitherMap(sattr, rattr, true, string(semconv.ContainerIDKey), string(semconv.K8SPodUIDKey))
+		}
```

Comment:
Prefer to move these to new util functions traceutil.GetOTelVersion & traceutil.GetOTelCID